### PR TITLE
Cleanup latexcontroller

### PR DESCRIPTION
**Pull request by jhilmer**
I have removed some `double` to `int` conversion compiler warnigns. Use constexpr and anonymous namespace for some const parameters.

### CODE REVIEW

File: src/core/control/LatexController.cpp [hunk 1]
```diff
@@ -115,16 +117,16 @@ void LatexController::findSelectedTexElement() {
         // this will get the position of the Latex properly
         EditSelection* theSelection = control->getWindow()->getXournal()->getSelection();
         xoj::util::Rectangle<double> rect = theSelection->getSnappedBounds();
-        this->posx = rect.x;
-        this->posy = rect.y;
+        this->posx = static_cast<int>(rect.x);
+        this->posy = static_cast<int>(rect.y);
```
Comment (bhennion) [inline src/core/control/LatexController.cpp:121]: All those static_casts feel off to me. Bluntly casting should never happen: we probably want to std::round, std::floor or std::ceil those values before.
In this case, I don't know which one it should be. This needs understanding the class a bit better.

Comment (jhilmer) [reply src/core/control/LatexController.cpp:121]: I this case the posx, posy, imgWidth, imgHeight should be double. They are used to in method call on TexImage there expect double.

Comment (jhilmer) [reply src/core/control/LatexController.cpp]: By changing the attributes to double the static_cast is mostly avoided.

File: src/core/control/LatexController.cpp [hunk 2]
```diff
@@ -109,8 +111,8 @@ void LatexController::findSelectedTexElement() {
     this->page = this->doc->getPage(pageNr);
     this->layer = page->getSelectedLayer();
 
-    this->selectedElem = view->getSelectedTex() != nullptr ? static_cast<Element*>(view->getSelectedTex()) :
-                                                             static_cast<Element*>(view->getSelectedText());
+    this->selectedElem = view->getSelectedTex() != nullptr ? dynamic_cast<Element*>(view->getSelectedTex()) :
+                                                             dynamic_cast<Element*>(view->getSelectedText());
```
[the remark this answers is not in the document]
Comment (jhilmer) [reply src/core/control/LatexController.cpp:115]: Will revert

File: src/core/control/LatexController.cpp [hunk 3]
```diff
@@ -42,8 +42,10 @@
 
 using std::string;
 
-const Color LIGHT_PREVIEW_BACKGROUND = Colors::white;
-const Color DARK_PREVIEW_BACKGROUND = Colors::black;
+namespace {
+constexpr Color LIGHT_PREVIEW_BACKGROUND = Colors::white;
+constexpr Color DARK_PREVIEW_BACKGROUND = Colors::black;
+}  // namespace
```
[the remark this answers is not in the document]
Comment (jhilmer) [reply src/core/control/LatexController.cpp:48]: Will remove the namespace.  Wow I learn something new.

**Review by bhennion — APPROVED** [review]
LGTM!
Squashing and merging in 24 hours unless an objection is raised.

**Pull request merged**